opal/mca/btl/iwarp/btl_iwarp.h

```diff
@@ -0,0 +1,927 @@
+typedef struct mca_btl_iwarp_srq_manager_t {
+    opal_mutex_t lock;
+    /* The keys of this hash table are addresses of
+       SRQs structures, and the elements are BTL modules
+       pointers that associated with these SRQs */
+    opal_hash_table_t srq_addr_table;
+} mca_btl_iwarp_srq_manager_t;
+
+struct mca_btl_iwarp_component_t {
+    mca_btl_base_component_3_0_0_t          super;  /**< base BTL component */
+
+    int                                ib_max_btls;
+    /**< maximum number of devices available to iwarp component */
+
+    int                                ib_num_btls;
+    /**< number of devices available to the iwarp component */
+
+    struct mca_btl_iwarp_module_t             **iwarp_btls;
+    /**< array of available BTLs */
+
+    opal_pointer_array_t devices; /**< array of available devices */
+    int devices_count;
+
+    int ib_free_list_num;
+    /**< initial size of free lists */
+
+    int ib_free_list_max;
+    /**< maximum size of free lists */
+
+    int ib_free_list_inc;
+    /**< number of elements to alloc when growing free lists */
+
+    opal_list_t                             ib_procs;
+    /**< list of ib proc structures */
+
+    opal_event_t                            ib_send_event;
+    /**< event structure for sends */
+
+    opal_event_t                            ib_recv_event;
+    /**< event structure for recvs */
+
+    opal_mutex_t                            ib_lock;
+    /**< lock for accessing module state */
+
+    char* ib_mpool_hints;
+    /**< hints for selecting an mpool component */
+
+    char *ib_rcache_name;
+    /**< name of ib registration cache */
+
+    uint8_t num_pp_qps;          /**< number of pp qp's */
+    uint8_t num_srq_qps;         /**< number of srq qp's */
+    uint8_t num_xrc_qps;         /**< number of xrc qp's */
+    uint8_t num_qps;             /**< total number of qp's */
+
+    opal_hash_table_t ib_addr_table; /**< used only for xrc.hash-table that
+                                       keeps table of all lids/subnets */
+    mca_btl_iwarp_qp_info_t* qp_infos;
+
+    size_t eager_limit;      /**< Eager send limit of first fragment, in Bytes */
+    size_t max_send_size;    /**< Maximum send size, in Bytes */
+    uint32_t max_hw_msg_size;/**< Maximum message size for RDMA protocols in Bytes */
+    uint32_t reg_mru_len;    /**< Length of the registration cache most recently used list */
+    uint32_t use_srq;        /**< Use the Shared Receive Queue (SRQ mode) */
+
+    uint32_t ib_cq_size[BTL_IWARP_MAX_CQ];  /**< Max outstanding CQE on the CQ */
+
+    int      ib_max_inline_data; /**< Max size of inline data */
+    unsigned int ib_pkey_val;
+    unsigned int ib_psn;
+    unsigned int ib_qp_ous_rd_atom;
+    uint32_t ib_mtu;
+    unsigned int ib_min_rnr_timer;
+    unsigned int ib_timeout;
+    unsigned int ib_retry_count;
+    unsigned int ib_rnr_retry;
+    unsigned int ib_max_rdma_dst_ops;
+    unsigned int ib_service_level;
```

Comment:
There may be no (relevant) pkeys, SLs, dynamic SLs, LMC, APM, multiple BTLs per LID, etc. in iWARP.  Should delete everything out of here that isn't relevant to iWARP (or will always be a fixed value just because it's in the verbs API but isn't relevant to iWARP).

Ditto for the module, modex, ...etc. structs.